docs/index.md

```diff
@@ -459,6 +459,531 @@ class Settings(BaseSettings):
     model_config = SettingsConfigDict(env_file='.env', extra='ignore')
 ```
 
+## Command Line Support
+
+Pydantic settings provides integrated CLI support, making it easy to quickly define CLI applications using Pydantic
+models. There are two primary use cases for Pydantic settings CLI:
+
+1. When using a CLI to override fields in Pydantic models.
+2. When using Pydantic models to define CLIs.
+
+By default, the experience is tailored towards use case #1 and builds on the foundations established in [parsing
+environment variables](#parsing-environment-variables). If your use case primarily falls into #2, you will likely want
+to enable [enforcing required arguments at the CLI](#enforce-required-arguments-at-cli).
+
+### The Basics
+
+To get started, let's look at a basic example for defining a Pydantic settings CLI:
+
+```py test="skip"
+from typing import List
+
+from pydantic import BaseModel
+
+from pydantic_settings import BaseSettings
+
+
+class DeepSubModel(BaseModel, use_attribute_docstrings=True):
+    """DeepSubModel class documentation."""
+
+    v4: List[int]
+    """the deeply nested sub model v4 option"""
+
+
+class SubModel(BaseModel, use_attribute_docstrings=True):
+    """SubModel class documentation."""
+
+    v1: int
+    """the sub model v1 option"""
+
+    deep: DeepSubModel
+    """The help summary for DeepSubModel and related options. This will be placed at top of group."""
+
+
+class Settings(BaseSettings, use_attribute_docstrings=True):
+    """The Settings class documentation will show in top level help text."""
+
+    v0: str
+    """the top level v0 option"""
+
+    sub_model: SubModel
+    """The help summary for SubModel related options. This will be placed at top of group."""
+
+
+Settings(_cli_prog_name='app', _cli_parse_args=['--help'])  # (1)!
```

Comment:
I would try to avoid using the `_` kwargs to `__init__`, and use class kwargs instead in most places in the docs - I think that approach is clearer and easier to understand.

Reply:
Sure, will update.